sfdoc/publish/tasks.py

```diff
@@ -10,294 +10,368 @@
 
 from .amazon import S3
 from .exceptions import SfdocError
-from .html import HTML
+from .html import HTML, collect_html_paths
 from .logger import get_logger
 from .models import Article
 from .models import Bundle
 from .models import Image
 from .models import Webhook
 from .salesforce import Salesforce
-from .logger import get_logger
-from .utils import is_html
-from .utils import skip_html_file
-from .utils import unzip
+from . import utils
+
+logger = None
+
+AWS_S3_PUBLISHED_HTML_REPOSITORY_URL = (
+    f"s3://{settings.AWS_S3_BUCKET}/{settings.AWS_S3_PUBLISHED_HTML_REPOSITORY_DIR}"
+)
+
+AWS_S3_DRAFT_HTML_REPOSITORY_URL = (
+    f"s3://{settings.AWS_S3_BUCKET}/{settings.AWS_S3_DRAFT_HTML_REPOSITORY_DIR}"
+)
+
+
+def _download_and_unpack_easydita_bundle(bundle, path):
+    logger.info("Downloading easyDITA bundle from %s", bundle.url)
+    auth = (settings.EASYDITA_USERNAME, settings.EASYDITA_PASSWORD)
+    response = requests.get(bundle.url, auth=auth)
+    zip_file = BytesIO(response.content)
+    with TemporaryDirectory(f"bundle_{bundle.pk}_") as tempdir:
+        utils.unzip(zip_file, tempdir, recursive=True, ignore_patterns=["*/assets/*"])
+
+        rootpath = utils.find_bundle_root_directory(tempdir)
+        assert os.path.exists(os.path.join(rootpath, "log.txt"))
+        utils.sync_directories(rootpath, path)
+        assert os.path.exists(os.path.join(path, "log.txt"))
+
 
+def _download_repository_state_from_s3(path):
+    assert os.path.exists(path)
+    logger.info(
+        "Downloading repository state from S3 %s", AWS_S3_PUBLISHED_HTML_REPOSITORY_URL
+    )
 
-def _process_bundle(bundle, path, enforce_no_duplicates = True):
+    # update our clean local copy
+    utils.s3_sync(AWS_S3_PUBLISHED_HTML_REPOSITORY_URL, settings.LOCAL_REPOSITORY_CACHE)
+    assert os.path.exists(settings.LOCAL_REPOSITORY_CACHE)
+
+    if len(os.listdir(settings.LOCAL_REPOSITORY_CACHE)):
+        assert os.path.exists(os.path.join(settings.LOCAL_REPOSITORY_CACHE, "log.txt"))
+
+    # copy it to the temporary working directory for this bundle
+    utils.sync_directories(settings.LOCAL_REPOSITORY_CACHE, path)
+    assert os.path.exists(path)
+    assert os.path.exists(os.path.join(settings.LOCAL_REPOSITORY_CACHE, "log.txt"))
+
+
+def _process_bundle(bundle, path, enforce_no_duplicates=True):
+    global logger
```

Comment:
Can we avoid using a global? Someday someone decides that it would be a good idea to process bundles in parallel threads and then kaboom

Reply:
b11bf8bd6c7e097f736262dcb53916e8d53b1ed2